### PR TITLE
Salt add azure volume support

### PR DESCRIPTION
**Pull request by flavianh**
Use
volumes:
- { size: 10 (default 100), lun: [0-15](default: 0), disk_label:
  <label>(default: <role-name>-disk-<lun>) }
  in the profile file to add a new volume

### CODE REVIEW

File: salt/cloud/clouds/msazure.py [hunk 1]
```diff
@@ -709,6 +742,95 @@ def wait_for_hostname():
...
+    Create and attach volumes to created node
+    '''
+    if call != 'action':
+        raise SaltCloudSystemExit(
+            'The create_attach_volumes action must be called with '
+            '-a or --action.'
+        )
+
+    if isinstance(kwargs['volumes'], str):
+        volumes = yaml.safe_load(kwargs['volumes'])
```
Comment (thatch45) [inline salt/cloud/clouds/msazure.py]: Good job! Almost everyone forgets to do a safe_load!

**Comment by thatch45** [comment]
https://jenkins.saltstack.com/job/salt-pr-lint-n/634/violations/file/salt/cloud/clouds/msazure.py/

Some minor lint erros, but I will clean them up for you @traxair, Thanks for the addition!

**Comment by thatch45** [comment]
Thanks for the addition!

File: salt/cloud/clouds/msazure.py (hunk 1 above)
Comment (traxair) [inline salt/cloud/clouds/msazure.py]: I copied this part from the aws file for consistency :)

Comment (thatch45) [inline salt/cloud/clouds/msazure.py]: You don't need to back out of taking credit! :)

**Comment by techhat** [comment]
@rallytime, please backport to 2015.2.

**Comment by traxair** [comment]
@rallytime it would be __awesome__ if this could get backported by tomorrow

**Comment by rallytime** [comment]
Okay, the backport to 2014.7 was reverted because it was in the wrong branch. This has now been backported to 2015.2 in #20215.

**Pull request merged**